exo/networking/manual/manual_discovery.py

```diff
@@ -0,0 +1,95 @@
+import asyncio
+import time
+import traceback
+from typing import List, Dict, Callable, Tuple
+from exo.networking.discovery import Discovery
+from exo.networking.peer_handle import PeerHandle
+from exo.topology.device_capabilities import DeviceCapabilities, DeviceFlops, device_capabilities, UNKNOWN_DEVICE_CAPABILITIES
+from exo.helpers import DEBUG, DEBUG_DISCOVERY
+from exo.networking.manual.read_config import ReadManualConfig
+from exo.networking.grpc.grpc_peer_handle import GRPCPeerHandle
+
+class ManualDiscovery(Discovery):
+  def __init__(
+    self,
+    node_id: str,
+    node_port: int,
+    create_peer_handle: Callable[[str, str, DeviceCapabilities], PeerHandle],
+    device_capabilities: DeviceCapabilities = UNKNOWN_DEVICE_CAPABILITIES,
+    discovery_config: str = "topology.yml",
+    discovery_interval: int = 5,
+    discovery_timeout: int = 30,
+    update_interval: int = 15,
+  ):
+    self.node_id = node_id
+    self.node_port = node_port
+    self.discovery_config = discovery_config
+    self.list_device = ReadManualConfig(discovery_config=self.discovery_config)
+    self.create_peer_handle = create_peer_handle
+    self.discovery_interval = discovery_interval
+    self.discovery_timeout = discovery_timeout
+    self.update_interval = update_interval
+    self.device_capabilities = device_capabilities
+    self.known_peers: Dict[str, Tuple[PeerHandle, float, float]] = {}
```

Comment:
Shouldn't this just be `Dict[str, PeerHandle]` now? I don't think we need the times any more.

Reply:
I don't know about that, for now I simplified the discovery with one function that input all the other devices on startup.
But I remove disconnection, ... etc for debug purpose.

Frankly it is your call, same for the GRPC problem.
Do you have some discord time to help me on : 
is_connected = await new_peer_handle.is_connected()
health_ok = await new_peer_handle.health_check()
that are not working, please ?

Thanks in advance.
Best Regards.
Benjamin.

Reply:
sure I can help. it's quite simple tho - only healthy peers should be in the topology